src/interface/fileList/fileList.js

```diff
@@ -23,8 +11,43 @@ async function fileList(req, resp) {
         return { ipfsHash, gatewayUrl, visibility, fileId, chainId, contractAddress };
     });
 
+    return fileList;
+}
+
+async function getUniqueFile(req, resp) {
+    const { ipfsHash } = req.query;
+    if (!ipfsHash) {
+        return resp.status(400).json({ error: 'ipfsHash is required' });
+    }
+
+    const file = await File.findOne(ipfsHash);
+    if (!file) {
+        return resp.status(400).json({ error: 'no files found for given ipfsHash' });
+    }
+
+    const fileList = getResponse([file])
+    resp.status(200).json(fileList);
+}
+
+async function fileList(req, resp) {
+    if (!req.isAuthenticated) {
+        return resp
+            .status(401)
+            .json({ error: 'UNAUTHORISED REQUEST' })
+    }
+    const { invokerAddress } = req.query;
+
+
+    const files = await File.findAll(invokerAddress);
+    if (!files) {
+        return resp
+            .status(400)
+            .json({ error: 'no files found for given invokerAddress' });
```

Comment:
maybe we should just send an empty list instead of an error with status code 400

Reply:
🤔
Alright, I'll add message that no files found. And keep the status 200.